x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,531 @@
+      },
+    })
+    .then((response) => {
+      log.info(`${runnerId} : Environment ${projectName} was successfully deleted...`);
+    })
+    .catch((error) => {
+      onEarlyExit(`${error.code}:${error.data}`);
+    });
+}
+
+// Method to reset the credentials for the created environment.
+async function resetCredentials(
+  baseUrl: string,
+  environmentId: string,
+  runnerId: string,
+  apiKey: string
+): Promise<Credentials> {
+  log.info(`${runnerId} : Reseting credentials`);
+  const credentials = {} as Credentials;
+
+  await poll(async () => {
+    return axios
+      .post(
+        `${baseUrl}/api/v1/serverless/projects/security/${environmentId}/_reset-credentials`,
+        {},
+        {
+          headers: {
+            Authorization: `ApiKey ${apiKey}`,
+          },
+        }
+      )
+      .then((response) => {
+        credentials.password = response.data.password;
+        credentials.username = response.data.username;
+      })
+      .catch((error) => {
+        throw new Error(`${error.code}:${error.data}`);
+      });
+  });
+  return credentials;
+}
+
+// Wait until elasticsearch status goes green
+async function waitForEsStatusGreen(esUrl: string, auth: string, runnerId: string) {
+  await poll(async () => {
+    await axios
+      .get(`${esUrl}/_cluster/health?wait_for_status=green&timeout=50s`, {
+        headers: {
+          Authorization: `Basic ${auth}`,
+        },
+      })
+      .then((response) => {
+        log.info(`${runnerId}: Elasticsearch is ready with status ${response.data.status}.`);
+      })
+      .catch((error) => {
+        if (error.code === 'ENOTFOUND') {
+          log.info(`${runnerId}: The elasticsearch url is not yet reachable. Retrying in 20s...`);
+        }
+        throw new Error(`${runnerId} - ${error.code}:${error.data}`);
+      });
+  });
+}
+
+// Wait until Kibana is available
+async function waitForKibanaAvailable(kbUrl: string, auth: string, runnerId: string) {
+  await poll(async () => {
+    await axios
+      .get(`${kbUrl}/api/status`, {
+        headers: {
+          Authorization: `Basic ${auth}`,
+        },
+      })
+      .then((response) => {
+        if (response.data.status.overall.level !== 'available') {
+          log.info(`${runnerId}: Kibana is not available. Retrying in 20s...`);
+          throw new Error(`${runnerId}: Kibana is not available. Retrying in 20s...`);
+        }
+      })
+      .catch((error) => {
+        if (error.code === 'ENOTFOUND') {
+          log.info(`${runnerId}: The kibana url is not yet reachable. Retrying in 20s...`);
+        }
+        throw new Error(`${runnerId} - ${error.code}:${error.data}`);
+      });
+  });
+}
+
+export const cli = () => {
+  run(
+    async () => {
+      log = new ToolingLog({
+        level: 'info',
+        writeTo: process.stdout,
+      });
+
+      const PROJECT_NAME_PREFIX = 'kibana-cypress-security-solution-ephemeral';
+      const QA_BASE_ENV_URL = 'https://global.qa.cld.elstc.co';
```

Comment:
Should this values be hardcoded here?

Reply:
For now, we can leave them as they are. When I get access to the vault I will clean this script. They are indeed hardcoded right now. But we will tackle this when this script runs for both PROD and generally uses vault in an appropriate way.